go-controller/pkg/ovn/admin_network_policy_test.go

```diff
@@ -1025,6 +1077,473 @@ var _ = ginkgo.Describe("OVN ANP Operations", func() {
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v6)
+				namedPortACLCount = 0
+				for _, acl := range acls {
+					acl := acl
+					if strings.Contains(acl.ExternalIDs[libovsdbops.PortPolicyProtocolKey.String()], "namedPort") {
+						namedPortACLCount++
+					}
+					expectedDatabaseState = append(expectedDatabaseState, acl)
+				}
+				// 3 ingress protocol ACLs
+				// 3 egress[0] protocol ACLs and 2 egress[0] namedPort ACLs,
+				// 3 egress[1] protocol ACLs
+				gomega.Expect(len(acls)).To(gomega.Equal(11))
+				gomega.Expect(namedPortACLCount).To(gomega.Equal(2))
+				gomega.Eventually(fakeOVN.nbClient).Should(libovsdbtest.HaveDataIgnoringUUIDs(expectedDatabaseState))
+
+				ginkgo.By("4. Delete matching peer pod and ensure ACLs are updated correctly")
+				anpPeerPod.ResourceVersion = "3"
+				err = fakeOVN.fakeClient.KubeClient.CoreV1().Pods(anpPeerPod.Namespace).Delete(context.TODO(), anpPeerPod.Name, metav1.DeleteOptions{})
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				delete(namedPorts, "web")
+				delete(namedPorts, "web123")
+				acls = getACLsForANPRules(anp)
+				pg = getDefaultPGForANPSubject(anp.Name, []string{t2.portUUID}, acls, false)
+				peerNSASIPv4, peerNSASIPv6 = buildNamespaceAddressSets(anpPeerNamespaceName, []string{}) // pod is gone from peer namespace address-set
+				expectedDatabaseState = []libovsdbtest.TestData{pg, subjectNSASIPv4, subjectNSASIPv6, peerNSASIPv4, peerNSASIPv6, peerNS2ASIPv4, peerNS2ASIPv6}
+				expectedDatabaseState = append(expectedDatabaseState, getExpectedDataPodsAndSwitches([]testPod{t2}, []string{node1Name})...)
+				expectedDatabaseState = append(expectedDatabaseState, peerASIngressRule0v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASIngressRule0v6)
+				peerASEgressRule0v4, peerASEgressRule0v6 = buildANPAddressSets(anp, 0, []string{}, libovsdbutil.ACLEgress)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule0v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule0v6)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v4)
+				expectedDatabaseState = append(expectedDatabaseState, peerASEgressRule1v6)
+				for _, acl := range acls {
+					acl := acl
+					expectedDatabaseState = append(expectedDatabaseState, acl)
+				}
+				p := func(item *nbdb.ACL) bool {
+					return strings.Contains(item.ExternalIDs[libovsdbops.PortPolicyProtocolKey.String()], "namedPort")
+				}
+				gomega.Eventually(func() (int, error) {
+					namedPortACLs, err := libovsdbops.FindACLsWithPredicate(fakeOVN.nbClient, p)
+					if err != nil {
+						return 100, fmt.Errorf("unable to find default ANP ACLs: %v", err)
+					}
+					namedPortACLCount = len(namedPortACLs)
+					return namedPortACLCount, nil
+				}, 2).Should(gomega.Equal(0))
```

Comment:
won't this be checked by `expectedData` comparison?

Reply:
yeah it will, so when writing code there was a bug in the way we compute the namedPorts ACLs hehe and because its tricky to also compute it in the UTs, for a second I was confused whether bug was in UT logic or real code :) so wrote this extra matching on both UT logic and main logic to be very explicit what is expected from each thing at this stage. I can also remove it if this seems confusing, but I see no harm in keeping it so didn't delete it, TBH I don't mind either ways

Reply:
I see unit test code as any other code we add and we have to maintain, so I think less is better. A simple example: if we ever change how acls are built for ANP, we could just update `getACLsForANPRules` code, all tests would stay exactly the same (only tests for getACLsForANPRules will have to change, if we have any). But if we insert extra checks for the expected data returned by getACLsForANPRules into every test, we will have to update every single test, even though the test logic itself didn't change.

Reply:
sure let me remove them like I said I don't mind either ways; I don't find it that important whether it stays or goes away, I'll delete it

Reply:
deleted. done